str.sh

```diff
@@ -0,0 +1,22 @@
+# Join concatenates the elements of `lst` into a new string
+# with values separated by `sep`. `Lst` must contains only
+# string values.
+# This function is the bastard brother of builtin `split`.
+fn join(lst, sep) {
```

Comment:
`lst` has to be a list right? Maybe add this to the comment. If not, ignore this comment.